### PR TITLE
Adding support for insecure parameter to allow access to https with invalid certs

### PR DESCRIPTION
**Pull request by leachuk**
Enabling optional support for expired ssl certs when using https. Was necessary to overcome an invalid cert on a clients development server while using jackrabbit-filevault.

### CODE REVIEW

**Comment by wildone** [comment]
+1

**Comment by kwin** [comment]
@leachuk Thanks for this PR. To enable to proceed with it please first create an according JIRA ticket and link it from your commit message. Also please rebase and fix the conflict and some issue related to HTTPS has been fixed in https://github.com/apache/jackrabbit/commit/3b2fae841179ee92bfe77a8c7d027fe104c97623 already.

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java [hunk 1]
```diff
@@ -529,16 +557,49 @@ private static Object getClientKey(SessionInfo sessionInfo) {
...
+                    SSLContext sslContext = SSLContextBuilder
+                            .create()
+                            .loadTrustMaterial(new TrustSelfSignedStrategy())
+                            .build();
+
+                    // we can optionally disable hostname verification.
+                    HostnameVerifier allowAllHosts = new NoopHostnameVerifier();
+
+                    // create an SSL Socket Factory to use the SSLContext with the trust self signed certificate strategy
+                    // and allow all hosts verifier.
```
Comment (kwin) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java]: what is the use case for this? Can we rather restrict that to accepting only self-signed certificates?

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java [hunk 2]
```diff
@@ -529,16 +557,49 @@ private static Object getClientKey(SessionInfo sessionInfo) {
...
+                    client = httpClientBuilder.build();
+                }
+
+                if (sessionInfo != null) {
+                    checkSessionInfo(sessionInfo);
+                    clients.put(clientKey, client);
+                    log.debug("Created Client " + client + " for SessionInfo " + sessionInfo);
+                }
+
+                } catch (KeyManagementException mke) {
```
Comment (kwin) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java]: all those exceptions should be wrapped in RepositoryExceptions

File: jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2davex/Spi2davexRepositoryServiceFactory.java [hunk 3]
```diff
@@ -40,6 +40,12 @@
      */
     public static final String PARAM_REPOSITORY_URI = "org.apache.jackrabbit.spi2davex.uri";
 
+    /**
+     * Optional configuration parameter to allow access to servers with invalid ssl certs.
```
Comment (kwin) [inline jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2davex/Spi2davexRepositoryServiceFactory.java]: You should clarify what invalid really means (expired, wrong host, self-signed, untrusted CA)

**Comment by Punkroku** [comment]
This message was for Konrad Windszus:

Are you a conrad I met in San Diego who taught me about Amtel Chips and wanted me to program robotics with you when I was practically broke? If so I made this logo for RAMA Robotics Arts Music and Animation at tiny.cc/ramalogo and I founded openInvent.club with my other ideas I cant afford. I also had a hackaday.io account.

It seems that my emails get "lost" you can snail mail me at the address below and I will reply with sticker gifts.(tiny.cc/oisticker)
Ember AR Leona
250 S. Lyon Ave Ste E 108
Hemet, CA 92543

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by wildone** [comment]
this one can be close as #88 supersedes this.

**Comment by leachuk** [comment]
Thanks for reviewing this. I've closed here as the PR has been superseded by https://github.com/apache/jackrabbit/pull/88 , which includes the new '--insecure' optional parameter.

**Pull request closed without merging**